variables.ad_admin.tf

```diff
@@ -0,0 +1,31 @@
+variable "mysql_ad_administrators" {
+  type = map(object({
+    server_id   = string
```

Comment:
when doing child variables - remember to consider which variables the module consumer has to supply vs those which are known to the module.  I'm looking at server_id 👀